csrc/scheduler/resize.cpp

```diff
@@ -0,0 +1,194 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+
+#include <debug.h>
+#include <instrumentation.h>
+#include <ir/utils.h>
+#include <scheduler/cache_policy_refiner.h>
+#include <scheduler/debug_utils.h>
+#include <scheduler/mark_aliases.h>
+#include <scheduler/pointwise_utils.h>
+#include <scheduler/registry_utils.h>
+#include <scheduler/resize.h>
+#include <scheduler/runtime_info.h>
+#include <scheduler/tools/inlining.h>
+#include <scheduler/tools/loop_domain_scheduler.h>
+#include <scheduler/tools/resize_utils.h>
+#include <val_graph_visitor.h>
+
+namespace nvfuser {
+
+namespace {
+
+// Just use the pointwise version for now
+TensorView* getReferenceTensor(Fusion* fusion) {
+  return pointwise_utils::getReferenceTensor(fusion);
+}
+
+} // namespace
+
+bool ResizeScheduler::canScheduleCompileTime(Fusion* fusion) {
+  if (!ir_utils::hasOpsOfType<SliceOp, PadOp>(fusion)) {
+    scheduler_debug_utils::canScheduleRejectReason(
+        schedulerType(), "No resize op to schedule");
+    return false;
+  }
+
+  if (scheduler_utils::isResharding(fusion)) {
+    scheduler_debug_utils::canScheduleRejectReason(
+        schedulerType(), "Fusion is resharding.");
+    return false;
+  }
+
+  if (ir_utils::hasAnyReductionOps(fusion)) {
+    scheduler_debug_utils::canScheduleRejectReason(
+        schedulerType(), "No support for reduction ops");
+    return false;
+  }
+
+  if (registry_utils::hasNonUniqueBcast(fusion)) {
+    scheduler_debug_utils::canScheduleRejectReason(
+        schedulerType(),
+        "Broadcasting dimension might be broadcasting to multiple sizes.");
+    return false;
+  }
+
+  // For now, the resize scheduler is only allowed for a limited set
+  // of fusion patterns. The restrictions are planned to be
+  // incrementally relaxed.
+
+  IdModel id_model(fusion, /*build_graphs=*/false);
+  const auto& broadcast_graph = id_model.buildBroadcastGraph();
+
+  // For now, only a single resize op is allowed to exist.
+  auto resize_based_tensor_ops = ir_utils::getOpsOfType<SliceOp, PadOp>(fusion);
+  if (resize_based_tensor_ops.size() != 1) {
+    scheduler_debug_utils::canScheduleRejectReason(
+        schedulerType(), "Only a single resize op is allowed.");
```

Comment:
nitpick. the error message sounds a bit confusing. we are rejecting because there's a single `slice` or `pad` op, but each could carry arbitrary number of resize with it.

something more like `Only a single resize-based tensor op is allowed.`

Reply:
nitpick. We can probably also move the id_model build graph after this check.

Reply:
Thanks for the feedback. I'll leave it as is since this is going to change soon.